zorkatest.js

```diff
@@ -0,0 +1,417 @@
+// prebuilt code
+const readline = require("readline");
+const { start } = require("repl");
+const rlInterface = readline.createInterface(process.stdin, process.stdout);
+
+function ask(questionText) {
+  return new Promise((resolve, reject) => {
+    rlInterface.question(questionText, resolve);
+  });
+}
+//start of my code
+class Item {
+  constructor(iName, iDescription, iAction, iTakeable, Idurability) {
+    this.name = iName;
+    this.description = iDescription;
+    this.takeable = iTakeable || false;
+    this.action = iAction || "nothing happens";
+    this.durability = Idurability;
+  }
+  take() {
+    //check if item takable and is in room's inventory
+    if (this.takeable && locationTable[curLoc].inventory.includes(this.name)) {
+      //add item to player inventory
+      playerInventory.push(this.name);
+      //remove item from the room's inventory
+      let takeIndex = locationTable[curLoc].inventory.indexOf(this.name);
+      locationTable[curLoc].inventory.splice(takeIndex, 1);
+      //return a confirmation value to user
+      return `you picked up ${this.name}`;
+    } else {
+      return `you are unable to pick that up`;
+    }
+  }
+  use() {
+    //console.log(this.durability);
+    if (playerInventory.includes(this.name)) {
+      if (this.durability > 0) {
+        this.durability--;
+        //console.log(this.durability)
+        if (this.name === `stick` && curLoc === `startRoom`) {
+          return `you wave to stick around, ready to take on the world`;
+        } else {
+          //console.log(this.durability)
+          return this.action;
+        }
+      } else if (this.durability <= 0) {
+        return `you have used this as much as you can`;
+      }
+    } else {
+      return `you don't have one of those to use`;
+    }
+  }
+
+  drop() {
+    //if its in your inventory remove it from you inventory and add it to your current location's inventory
+    if (playerInventory.includes(this.name)) {
+      //add it to the room's inventory
+      locationTable[curLoc].inventory.push(this.name);
+      //remove it from the player's inventory
+      let dropIndex = playerInventory.indexOf(this.name);
+      playerInventory.splice(dropIndex, 1);
+      return `you dropped ${this.name},`;
+    } else {
+      return `you don't have one of those to drop`;
+    }
+  }
+}
+class Room {
+  constructor(name, description, inventory, connectingRooms) {
+    this.location = name;
+    this.description = description;
+    this.inventory = inventory;
+    this.conRooms = connectingRooms;
+  }
+}
+
```

Comment:
Your organization is good in this file. I would like to see a visual break that at the start of a new set of data. In this instance, it would separate classes from variable declarations. A commonly used example would be to start a new section with a line of commented out a line of asterisks: //*********************CLASSES******************** or //**************VAR DECLARATIONS**********